zhmccli/_cmd_partition.py

```diff
@@ -916,6 +967,133 @@ def cmd_partition_console(cmd_ctx, cpc_name, partition_name, options):
             "{exc}: {msg}".format(exc=exc.__class__.__name__, msg=exc))
 
 
+def cmd_partition_dump(cmd_ctx, cpc_name, partition_name, **options):
+    # pylint: disable=missing-function-docstring
+
+    client = zhmcclient.Client(cmd_ctx.session)
+    partition = find_partition(cmd_ctx, client, cpc_name, partition_name)
+
+    name_map = {
+        # These parameters are the same for CPCs with and without the storage
+        # management feature. Feature-specific parameters are added further
+        # down:
+        'configuration': 'dump-configuration-selector',
+        'parameters': 'dump-os-specific-parameters',
+        'lba': 'dump-record-lba',
+
+        # Options that are handled in this function:
+        'volume': None,
+        'timeout': None,
+    }
+    options = original_options(options)
+    properties = options_to_properties(options, name_map)
+    volume_opt = options['volume']  # It is required
+    timeout = options['timeout']  # It is optional but has a default
+
+    if storage_management_feature(partition):
+
+        # Check and parse the --volume option into SG.SV or UUID
+        volume_parts = volume_opt.split('.')
+        if len(volume_parts) == 1:  # format: UUID
+            uuid = volume_parts[0]
+            sg_list = partition.list_attached_storage_groups()
+            storage_volume = None
+            for sg in sg_list:
+                try:
+                    storage_volume = sg.storage_volumes.find(uuid=uuid)
+                    break
+                except zhmcclient.NotFound:
+                    continue
+            if not storage_volume:
+                raise_click_exception(
+                    "Storage volume with UUID '{uuid}' specified in the "
+                    "--volume option was not found in any storage group "
+                    "attached to partition '{part}' of CPC '{cpc}'.".
+                    format(uuid=uuid, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
+        elif len(volume_parts) == 2:  # format: SG.SV
+            sg_name, sv_name = volume_parts
+            sg_list = partition.list_attached_storage_groups()
+            sg = None
+            for _sg in sg_list:
+                if _sg.name == sg_name:
+                    sg = _sg
+                    break
+            if not sg:
+                raise_click_exception(
+                    "Storage group '{sg}' specified in the --volume option was "
+                    "not found in the storage groups attached to partition "
+                    "'{part}' of CPC '{cpc}'.".
+                    format(sg=sg_name, part=partition_name, cpc=cpc_name),
+                    cmd_ctx.error_format)
```

Comment:
Maybe a little bit overkill...
```suggestion
try:
  sg = next(filter(lambda sg_ele: sg_ele.name == sg_name, sg_list))
except StopIteration:
  raise ...
```

Reply:
This changes 6 easy to understand lines into 3 lines one of which is much harder to understand.

At the end of the day, it is a matter of whether or not one likes using `filter()`. Shorter code and seemingly more elegant, but requires me every time to think for a minute to understand what it does. I can grasp what the original code does, within seconds. Python introduced list comprehensions because users found it hard to understand combinations of filter/map. So I'm not the only one in this camp it seems ;-)

I prefer using the original code.
REJECTED.